src/semantics/value_types.ts

```diff
@@ -654,6 +654,7 @@ export class EnumType extends ValueType {
         public members: Map<string, string | number>,
     ) {
         super(ValueTypeKind.ENUM, typeId);
+        if (memberType.isPrimitive) this.setPrimitive();
```

Comment:
Enum type is different with primitive type, I think even the member type is primitive, the enum type can not be set to primitive.

Reply:
So we regard **_Enum_** as a specific independent type, not its **_memberType_** type.

Reply:
Yeah, regard `Enum` as an independent type is more resonable.